pmlc/dialect/stripe/types.h

```diff
@@ -55,7 +55,7 @@ struct TensorDim {
   int64_t stride;
 
   // The hardware class identified by this dimension.
-  llvm::StringRef cls;
+  mlir::Identifier cls;
```

Comment:
I can't default constructor TensorDim when this is an identifier. You can add a default ctor that sets this value to `"address"` or some other reasonable value.

Reply:
Oh, I see now why this isn't possible (or good), we need an MLIR context to create an identifier, and without it being an identifier, we have tons of copies. Oh, this is why I was thinking a `StringRef` would be safe, as long as the StringRef always came from an identifier.

Reply:
I can't -- `Identifier::get()` requires a context.  :-/  Thoughts?  (Maybe this should go back to being an uninterned string?)

Reply:
It's OK as is I think (where the caller must construct a TensorDim with a valid identifier).